src/character.cpp

```diff
@@ -566,14 +566,17 @@ std::vector<const item *> Character::items_with( const std::function<bool(const
 
 item& Character::i_add(item it)
 {
- itype_id item_type_id = "null";
- if( it.type ) item_type_id = it.type->id;
+    itype_id item_type_id = "null";
+    if( it.type ) {
+        item_type_id = it.type->id;
+    }
 
- last_item = item_type_id;
+    last_item = item_type_id;
```

Comment:
You can get rid of that local variable and assign `it.typeId()` directly. `item::type` should never be null, and even if, `typeId()` would take care of that.